libs/langchain-google-common/src/output_parsers.ts

```diff
@@ -0,0 +1,266 @@
+import { BaseLLMOutputParser } from "@langchain/core/output_parsers";
+import { Callbacks } from "@langchain/core/callbacks/manager";
+import { ChatGeneration, Generation } from "@langchain/core/outputs";
+import { MessageContent } from "@langchain/core/messages";
+import {
+  GeminiGroundingChunk,
+  GeminiGroundingMetadata,
+  GeminiGroundingSupport,
+} from "./types.js";
+
+type Generations = Generation[] | ChatGeneration[];
+
+type GroundingInfo = {
+  metadata: GeminiGroundingMetadata;
+  supports: GeminiGroundingSupport[];
+};
+
+export abstract class BaseGoogleSearchOutputParser extends BaseLLMOutputParser<string> {
+  lc_namespace: string[] = ["google_common", "output_parsers"];
+
+  generationToGroundingInfo(
```

Comment:
Should we make all these protected too?

Makes e.g. API refs hard to navigate if the user isn't intended to call them directly

Reply:
I'm fine with all of them protected. I tended to think to make them open if people did want to use them natively, but I can't imagine that being important.

I'll take care of it later if you don't get to it.